client/python/lib/docs/grpc_client.md

```diff
@@ -0,0 +1,138 @@
+
+## <kbd>class</kbd> `GrpcClient`
+
+---
+
+<a href="../../../../client/python/lib/ovmsclient/tfs_compat/grpc/serving_client.py#L61"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>method</kbd> `get_model_metadata`
+
+```python
+get_model_metadata(request)
+```
+
+Send `GrpcModelMetadataRequest` to the server and return response.
+
+
+**Args:**
+ 
+ - <b>`request`</b>:  `GrpcModelMetadataRequest` object. 
+
+
+**Returns:**
+ `GrpcModelMetadataResponse` object 
+
+
+**Raises:**
+ 
+ - <b>`TypeError`</b>:   if request argument is of wrong type.
+ - <b>`ValueError`</b>:   if request argument has invalid contents.
+ - <b>`ConnectionError`</b>:   if there was an error while sending request to the server.
+
+
+**Examples:**
+ 
+```python
+
+config = {
+    "address": "localhost",
+    "port": 9000
+}
+client = make_grpc_client(config)
+request = make_model_metadata_request("model")
+response = client.get_model_metadata(request)
+
+```
+
+---
+
+<a href="../../../../client/python/lib/ovmsclient/tfs_compat/grpc/serving_client.py#L89"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>method</kbd> `get_model_status`
+
+```python
+get_model_status(request)
+```
+
+Send `GrpcModelStatusRequest` to the server and return response. 
+
+
+**Args:**
+ 
+ - <b>`request`</b>:  `GrpcModelStatusRequest` object. 
+
+
+**Returns:**
+ `GrpcModelStatusResponse` object 
+
+
+**Raises:**
+ 
+ - <b>`TypeError`</b>:   if request argument is of wrong type.
+ - <b>`ValueError`</b>:   if request argument has invalid contents.
+ - <b>`ConnectionError`</b>:   if there was an error while sending request to the server.
+
+
+**Examples:**
+
+```python
+
+config = {
+    "address": "localhost",
+    "port": 9000
+}
+client = make_grpc_client(config)
+request = make_model_status_request("model")
+response = client.get_model_status(request)
+
+```
+
+
+---
+
+<a href="../../../../client/python/lib/ovmsclient/tfs_compat/grpc/serving_client.py#L33"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>method</kbd> `predict`
+
+```python
+predict(request)
+```
+
+Send `GrpcPredictRequest` to the server and return response. 
+
+
+**Args:**
+ 
+ - <b>`request`</b>:  `GrpcPredictRequest` object. 
+
+
+**Returns:**
+ `GrpcPredictResponse` object 
+
+
+**Raises:**
+ 
+ - <b>`TypeError`</b>:   if request argument is of wrong type.
+ - <b>`ValueError`</b>:   if request argument has invalid contents.
+ - <b>`ConnectionError`</b>:   if there was an error while sending request to the server.
+
+
+**Examples:**
+
+```python
+
+config = {
+"address": "localhost",
+"port": 9000
+}
+client = make_grpc_client(config)
+request = make_predict_request({"input": [1, 2, 3]}, "model")
```

Comment:
what will be the name of the predict request with REST API?

Reply:
I was thinking of 2 ways to handle that:
 1. Pair of methods: `make_grpc_predict_request` and `make_http_predict_request`
 2. Parametrize this method like: `make_predict_request(data, api_type)`

Do you have any thoughts on that?

Reply:
https://github.com/openvinotoolkit/model_server/pull/848 introduces exported API. Such methods will be available via:
`ovmsclient.make_grpc_predict_request()` / `ovmsclient.make_http_predict_request()`
and
`ovmsclient.grpc.make_predict_request()` / `ovmsclient.http.make_predict_request()`